### PR TITLE
Adding new params to create_cluster function

### PR DESCRIPTION
**Pull request by rocknes**
Testing done :
CLI command :
(vcd-cli) C:\code\vcd-cli>vcd cluster create -n net1 -N 2 -c 2 -m 1025 --ssh-key ssh-key.txt "aritra-test-cluster"

cse.log :
DEBUG    2017-11-29 08:14:08,615 container_service_extension.processor    process_request                      55  : request body: {"name": "aritra-test-cluster", "node_count": 2, "vdc": "vdc1", "cpu-count": "2", "memory": "1025", "network": "net1", "storage_profile": null, "ssh-key": "sample SSH key"}

Note : the command failed with an error stating 'not enough IP addresses' on the extension. But I believe it's not related to my changes.

### CODE REVIEW

File: pyvcloud/vcd/cluster.py [hunk 1]
```diff
@@ -40,12 +40,24 @@ def get_clusters(self):
...
-                       storage_profile=None):
+                       cpu_count=None, memory=None, storage_profile=None,
+                       ssh_key=None):
+        """
+        Create a new Kubernetes cluster
+        :param cpu_count: (str): The number of virtual cpus on each of the nodes in the cluster
+        :param memory: (str): The amount of memory (in MB) on each of the nodes in the cluster
+        :param ssh_key: (str): The ssh key that clients can use to log into the node vms without 
+                               explicitly providing passwords
+        :return: (json) A paresed json object describing the requested cluster.
```
Comment (sahithi) [inline pyvcloud/vcd/cluster.py]: parsed spelling check

File: pyvcloud/vcd/cluster.py [hunk 2]
```diff
@@ -40,12 +40,24 @@ def get_clusters(self):
             raise Exception(json.loads(response.content))
 
     def create_cluster(self, vdc, network_name, name, node_count=2,
-                       storage_profile=None):
+                       cpu_count=None, memory=None, storage_profile=None,
+                       ssh_key=None):
+        """
+        Create a new Kubernetes cluster
+        :param cpu_count: (str): The number of virtual cpus on each of the nodes in the cluster
```
Comment (sahithi) [inline pyvcloud/vcd/cluster.py]: Describe other existing params as well.

File: pyvcloud/vcd/cluster.py (hunk 1 above)
Comment (rocknes) [reply pyvcloud/vcd/cluster.py]: Done

File: pyvcloud/vcd/cluster.py (hunk 2 above)
Comment (rocknes) [reply pyvcloud/vcd/cluster.py]: Done

File: pyvcloud/vcd/cluster.py [hunk 3]
```diff
@@ -40,12 +40,24 @@ def get_clusters(self):
...
+        :param memory: (str): The amount of memory (in MB) on each of the nodes in the cluster
+        :param ssh_key: (str): The ssh key that clients can use to log into the node vms without 
+                               explicitly providing passwords
+        :return: (json) A paresed json object describing the requested cluster.
+        """  # NOQA
         method = 'POST'
         uri = self._uri
         data = {'name': name, 'node_count': node_count, 'vdc': vdc,
+                'cpu-count': cpu_count,
```
Comment (pacogomez) [inline pyvcloud/vcd/cluster.py]: field names in json documents should not contain dashes, it makes harder to convert to python objects. `cpu_count` is preferred

File: pyvcloud/vcd/cluster.py [hunk 4]
```diff
@@ -40,12 +40,24 @@ def get_clusters(self):
...
+                'memory': memory,
                 'network': network_name,
-                'storage_profile': storage_profile}
+                'storage_profile': storage_profile,
+                'ssh-key': ssh_key}
```
Comment (pacogomez) [inline pyvcloud/vcd/cluster.py]: field names in json documents should not contain dashes, it makes harder to convert to python objects. `ssh_key` is preferred

File: pyvcloud/vcd/cluster.py (hunk 3 above)
Comment (rocknes) [reply pyvcloud/vcd/cluster.py]: Fixed

File: pyvcloud/vcd/cluster.py (hunk 4 above)
Comment (rocknes) [reply pyvcloud/vcd/cluster.py]: Fixed

**Pull request merged**